File: 819.Most_Common_Word.cpp
def mostCommonWord(self, paragraph, banned):
        """
        :type paragraph: str
        :type banned: List[str]
        :rtype: str
        """
        return collections.Counter(i.lower() for i in re.split("\W+", paragraph) if i and i.lower() not in banned).most_common(1)[0][0]

File: 819._Most_Common_Word.py
class Solution:
    def mostCommonWord(self, paragraph, banned):
        banned=set(banned)
        count = collections.defaultdict(int)
        for word in paragraph.lower().split():
            temp=word.strip("!?',;.")
            # print(temp)
            count[temp]+=1
        result=0
        max_word=''
        for word in count:
            if count[word]>result and word not in banned:
                max_word=word
                result=count[word]
        return max_word